src/servant/tt_001/ghost.c

```diff
@@ -375,62 +390,67 @@ void UpdateServantDefault(Entity* self) {
     s32 temp_s2;
     s32 temp_s1;
 
-    g_api.func_8011A3AC(self, 0, 0, &D_us_80173810);
+    g_api.func_8011A3AC(self, 0, 0, &s_GhostStats);
     if (g_IsServantDestroyed) {
         self->zPriority = PLAYER.zPriority - 2;
     }
     self->hitboxWidth = 0;
     self->hitboxHeight = 0;
+
+    // I think if step is < 2, the target the ghost is moving to is Alucard
     if (self->step < 2) {
+        // I don't believe this is the correct flag.  This appears to be a flag
+        // with the foreground layer, not the stage.
         if (D_8003C708.flags & STAGE_INVERTEDCASTLE_FLAG) {
```

Comment:
Maybe? The main purpose seems to be loading graphics https://github.com/Xeeynamo/sotn-decomp/blob/6301e7d7cec9919bef5da074ee63eb934b0ea605/src/dra/5087C.c#L1664

This place seems odd since it refers to the succubus, so maybe it's not inverted castle related. https://github.com/Xeeynamo/sotn-decomp/blob/6301e7d7cec9919bef5da074ee63eb934b0ea605/src/dra/5087C.c#L997

Xeeynamo would know better

Reply:
Used in this context though, we're setting the target location that the ghost is traveling to.  If that flag is for inverted castle, it essentially hard codes the ghost's target location.  That and ServantUnk0 does some logic that does switch if we are in the inverted castle.  It seems really weird in this context and is probably something to look into at some point, but for now, I just wanted to document it once we have a better handle on what that FgFlags struct is used for elsewhere.

I'm trying to think of a point in the game where the ghost would want to be essentially hard coded to go to the center of the screen, but I'm not sure

Reply:
Ok, I'll add an unknown flag and document that it's the same number as inverted castle flag, but used in a different context.  That way we could track it back if it turns out it should be inverted castle.

Reply:
I added a couple of flags for where that flag value gets set to the rect.params of the layer definition, but I didn't go replace the hex values everywhere that the flag is checked.  That's probably out of scope for this PR